lib/modules/manager/nix/extract.ts

```diff
@@ -1,23 +1,132 @@
+import { logger } from '../../../logger';
+import { getSiblingFileName } from '../../../util/fs';
 import { regEx } from '../../../util/regex';
 import { GitRefsDatasource } from '../../datasource/git-refs';
-import { id as nixpkgsVersioning } from '../../versioning/nixpkgs';
 import type { PackageDependency, PackageFileContent } from '../types';
+import { NixFlakeLock } from './schema';
 
-const nixpkgsRegex = regEx(/"github:nixos\/nixpkgs(\/(?<ref>[a-z0-9-.]+))?"/i);
+// TODO: add support to update nixpkgs branches in flakes.nix using nixpkgsVersioning
+
+// as documented upstream
+// https://github.com/NixOS/nix/blob/master/doc/manual/source/protocols/tarball-fetcher.md#gitea-and-forgejo-support
+const lockableHTTPTarballProtocol = regEx(
+  '^https://(.+)/(.+)/(.+)/archive/(.+).tar.gz$',
```

Comment:
```suggestion
  '^https://([^/]+)/([^/]+)/([^/]+)/archive/(.+).tar.gz$',
```
you don't want to match slashes in those groups.

are you using them later? then please use named capture groups.